v3/src/utilities/view-utils.ts

```diff
@@ -100,3 +100,12 @@ export const isWithinBounds = (dimension: number, rect?: DOMRect) => {
   const viewportRect = viewportEl.getBoundingClientRect()
   return (rect.right + dimension + 10) < viewportRect?.right
 }
+
+export const getPaletteTopPosition = (tempTop: number, paletteHeight: number, pointerMid: number) => {
+  const top = pointerMid-paletteHeight/2
+  if ((tempTop+top) > kCodapAppHeader + kGap) {
```

Comment:
My suggestion of adding `position: relative` to `.codap-container` was based on the assumption that the issue was that we had to position the palette in window coordinates. I now see that the issue is that `tempTop` and `pointerMid` are in window coordinates. I'm wondering if at some point it would make more sense to pass DOM elements around rather than coordinates in different coordinate systems, but probably not worth doing anything about it right now.